server/src/controllers/global_data_sources.controller.ts

```diff
@@ -72,6 +77,15 @@ export class GlobalDataSourcesController {
       throw new ForbiddenException('You do not have permissions to perform this action');
     }
 
+    if (kind === 'grpc') {
+      const rootDir = process.cwd().split('/').slice(0, -1).join('/');
+      const protoFilePath = `${rootDir}/protos/service.proto`;
+      const fs = require('fs');
+
+      const filecontent = fs.readFileSync(protoFilePath, 'utf8');
+      const rcps = await getServiceAndRpcNames(filecontent);
+      options.find((option) => option['key'] === 'protobuf').value = JSON.stringify(rcps, null, 2);
```

Comment:
I'd suppose when we add in upload capability these, preprocessing code would go onto the client side right?

Reply:
Yes